scene/3d/collision_object_3d.cpp

```diff
@@ -421,7 +474,7 @@ void CollisionObject3D::shape_owner_remove_shape(uint32_t p_owner, int p_shape)
 	ERR_FAIL_COND(!shapes.has(p_owner));
 	ERR_FAIL_INDEX(p_shape, shapes[p_owner].shapes.size());
 
-	const ShapeData::ShapeBase &s = shapes[p_owner].shapes[p_shape];
+	ShapeData::ShapeBase &s = shapes[p_owner].shapes.write[p_shape];
```

Comment:
Is this needed? It doesn't look like `ShapeBase` is modified.

Reply:
Let me know if that const_cast is good or there is another shortcut.

Reply:
Oh, my bad. I didn't realize it was needed for the call to `Object::disconnect`. In this case it's cleaner to set it back to what you did before, sorry.

It would actually be better for shapes to use `LocalVector` instead of `Vector` so writing doesn't involve a copy, but it's definitely not for this PR to fix that :)